dremio_toolkit/rebuild_metadata.py

```diff
@@ -0,0 +1,129 @@
+import json
+from dremio_toolkit.utils import Utils
+from dremio_toolkit.env_api import EnvApi
+from dremio_toolkit.logger import Logger
+from dremio_toolkit.rebuild_metadata_thread import RebuildMetadataThread
+import threading
+
+
+def parse_args():
+    # Process arguments
+    arg_parser = argparse.ArgumentParser(
+        description='rebuild_metadata is a part of the Dremio Toolkit. It forces Dremio to forget and then refresh metadata for all physical data sets as per supplied arguments.',
+        epilog='Copyright UCE Systems Corp. For any assistance contact developer at dremio@ucesys.com'
+    )
+    arg_parser.add_argument("-d", "--dremio-environment-url", help="URL to Dremio environment.", required=True)
+    arg_parser.add_argument("-u", "--user", help="User name. User must be a Dremio admin.", required=True)
+    arg_parser.add_argument("-p", "--password", help="User password.", required=True)
+    arg_parser.add_argument("-s", "--datasource", help="Limits the scope of the metadata refresh to physical datasets in a specified datasource. If not specified, metadata for all physical datasets in all datasources will be refreshed.", required=False)
+    arg_parser.add_argument("-c", "--concurrency", help="Concurrency for executing metadata refresh. It is not recommended to set it higher than 4 if dremio.iceberg.enabled is not set to True. Default concurrency is 1.", required=False, default=1)
+    arg_parser.add_argument("-r", "--report-filename", help="CSV file name for the JSON exception' report.", required=False)
+    arg_parser.add_argument("-l", "--log-level", help="Set Log Level to DEBUG, INFO, WARN, ERROR.",
+                            choices=['ERROR', 'WARN', 'INFO', 'DEBUG'], default='WARN')
+    arg_parser.add_argument("-f", "--log-filename", help="Set Log to write to a specified file instead of STDOUT.",
+                            required=False)
+    parsed_args = arg_parser.parse_args()
+    if parsed_args.report_filename is None:
+        print("report-filename argument has not been specified. Exception report will not be produced.")
+    if parsed_args.datasource is None:
+        print("datasource argument has not been specified. Metadata for all physical datasets in all data sources will be rebuilt.")
+    if int(parsed_args.concurrency) > 4:
+        print("Specifying concurrency higher than 4 without setting dremio.iceberg.enabled to True may impact Dremio performance and potentially bring Dremio cluster down.")
+    return parsed_args
+
+
+def rebuild_metadata(dremio_environment_url, user, password, datasource, concurrency, report_filename, log_level, log_filename):
+    logger = Logger(level=log_level, log_file=log_filename)
+    logger.new_process_status(1, 'Retrieving list of PDS for rebuilding metadata ...')
+    env_api = EnvApi(dremio_environment_url, user, password, logger)
+    pds_list = get_pds_list(env_api, datasource)
+    if pds_list is None:
+        logger.fatal("Unable to retrieve list of PDS. ")
```

Comment:
rebuilding metadata and spawning threads only makes sense if pds_list is not None, shouldn't we place it under `else` statement? ie:
```
    if pds_list is not None:
        <rebuild metadata code>...
    else:
        logger.fatal("Unable to retrieve list of PDS. ")
     ```

Reply:
fatal() will exit the application:
    def fatal(self, message: str, catalog: str = None) -> None:
        self._root_logger.critical(self._enrich_message(message, catalog))
        raise RuntimeError("Critical message: " + str(message))